File: WordPressCom-Stats-iOS/StatsGroup.m
#import "StatsGroup.h"

@implementation StatsGroup

- (NSUInteger)numberOfRows
{
    NSUInteger itemCount = 0;
    
    for (StatsItem *item in self.items) {
        itemCount += [item numberOfRows];
    }
    
    return itemCount;
}

- (StatsItem *)statsItemForTableViewRow:(NSInteger)row
{
    NSInteger index = row - self.offsetRows;
    
    NSInteger currentIndex = 0;
    
    // TODO - This doesn't account for nesting beyond one level
    for (StatsItem *item in self.items) {
        if (currentIndex == index) {
            return item;
        }
        
        if (item.isExpanded == YES) {
            for (StatsItem *subItem in item.children) {
                currentIndex++;
                
                if (currentIndex == index) {
                    return subItem;
                }
            }
        }
        
        currentIndex++;
    }
    
    return nil;
}

@end

File: WordPressCom-Stats-iOS/StatsGroup.swift
import Foundation

class StatsGroup {
    var items: Array<StatsItem>
    var titlePrimary: String
    var titleSecondary: String
    var icon: UIImage?
    
    init(items: Array<StatsItem>, titlePrimary: String, titleSecondary: String, icon: UIImage?) {
        self.items = items
        self.titlePrimary = titlePrimary
        self.titleSecondary = titleSecondary
        self.icon = icon
    }
}